File: LeetCode/1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int MAX = 0;
        for (int candy : candies)
            MAX = max(candy, MAX);
        vector<bool> result;
        for (int candy : candies)
            result.push_back(candy + extraCandies >= MAX);
        return result;
    }
};

File: LeetCode/1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.ts
function kidsWithCandies(candies: number[], extraCandies: number): boolean[] {
    const max_candies = Math.max(...candies)
    const n = candies.length;
    const result = [];

    for (let i = 0; i < n; i++) {
        if (candies[i] + extraCandies >= max_candies) {
            result.push(true)
        } else{
            result.push(false)
        }
    }

    return result;
};